src/main/java/com/theelixrlabs/healthcare/repository/DoctorRepository.java

```diff
@@ -1,13 +1,19 @@
 package com.theelixrlabs.healthcare.repository;
 
+import com.theelixrlabs.healthcare.constants.DoctorConstants;
 import com.theelixrlabs.healthcare.model.DoctorModel;
 import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.data.mongodb.repository.Query;
 import org.springframework.stereotype.Repository;
+import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
 
 @Repository
 public interface DoctorRepository extends MongoRepository<DoctorModel, UUID> {
 
     Optional<DoctorModel> findByAadhaarNumber(String AadhaarNumber);
+
+    @Query(DoctorConstants.SEARCH_DOCTOR_BY_NAME_QUERY)
```

Comment:
Please make use of findByName() instead of using query

Reply:
The requirement of this endpoint is to retrieve doctors whose first or last name starts with specific letters. For example, if I provide "J," it should retrieve all the first names and last names starting with "J" or "j". This requirement cannot be fulfilled with the findByName() method, so I used a custom query.